homework/list.js

```diff
@@ -1,79 +1,135 @@
+	 *
+	 *@constructor
+	 *@this {TodoListManager}
+	 */
+    var TodoListManager = function() {
+        /** @private */this._listNoteContent = [];
+        /** @private */this._wrapper = null;
+    };
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
+	/**
+	 *Write down the list.
+	 *
+	 *@this {TodoListManager}
+	 */
+    TodoListManager.prototype.drawList = function() {
+        var list = this._listNoteContent;
+        var ul = document.createElement('ul');
+        ul.id = 'note-title-list';
+        var buff = document.createDocumentFragment();
+        list.forEach(function(note, i) {
+            var li = document.createElement('li');
+            li.dataset.noteId = i;
+            li.classList.add('note-title');
+            li.textContent = note.title;
+            // Note: buff is captured, so we now have a
+            // little closure naturally.
+            buff.appendChild(li);
+        });
+        ul.appendChild(buff);
+        this._wrapper.appendChild(ul);
+    };
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
+	/**
+	 *While opening a note, here will be done.
+	 *
+	 *@param {object}
+	 *@this {TodoListManager}
+	 */
+    TodoListManager.prototype.onNoteOpen = function(event) {
+        if (event.target.classList.contains('note-title')) {
+            var id = event.target.dataset.noteId;
+            var content = this._listNoteContent[id];
+            window.dispatchEvent(new CustomEvent('note-open', {
+                detail: content
+            }));
+        };
+    };
+	
+	/**
+	 *When the html open, the first/default passage will open.
+	 *
+	 *@this {TodoListManager}
+	 */
+    TodoListManager.prototype.preloadFirstNote = function() {
+        if (this._listNoteContent.length !== 0) {
+            var content = this._listNoteContent[0];
+            window.dispatchEvent(new CustomEvent('note-open', {
+                detail: content
+            }));
+        }
+    };
+	
+	/**
+	 *To store the list data.
+	 *
+	 *@param {string} the list data.
+	 *@this {TodoListManager}
+	 */
+    TodoListManager.prototype.updateList = function(list) {
+        this._listNoteContent = list;
     };
-  }
 
-  function preloadFirstNote() {
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
+	/**
+	 *To get the data(passage) from demo-list-notes.json.
+	 *
+	 *@return {Promise} work flow
+	 *@this {TodoListManager}
+	 */
+    TodoListManager.prototype.fetchList = function(afterFetch) {
```

Comment:
Since you have successfully "Promisifed" this method, you don't need the callback anymore.